sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/PaginationCarbonReader.java

```diff
@@ -0,0 +1,276 @@
+      rowCountInSplits.add(sum);
+    }
+  }
+
+  /**
+   * Pagination query with from and to range.
+   *
+   * @param from must be greater than 0 and <= to
+   * @param to must be >= from and not outside the total rows
+   * @return array of rows between from and to (inclusive)
+   * @throws Exception
+   */
+  public Object[] read(long from, long to) throws IOException, InterruptedException {
+    if (isClosed) {
+      throw new RuntimeException("Pagination Reader is closed. please build again");
+    }
+    if (from < 1) {
+      throw new IllegalArgumentException("from row id:" + from + " is less than 1");
+    }
+    if (from > to) {
+      throw new IllegalArgumentException(
+          "from row id:" + from + " is greater than to row id:" + to);
+    }
+    if (to > getTotalRows()) {
+      throw new IllegalArgumentException(
+          "to row id:" + to + " is greater than total rows:" + getTotalRows());
+    }
+    return getRows(from, to);
+  }
+
+  /**
+   * Get total rows in the folder.
+   * It is based on the snapshot of files taken while building the reader.
+   *
+   * @return total rows from all the files in the reader.
+   */
+  public long getTotalRows() {
+    if (isClosed) {
+      throw new RuntimeException("Pagination Reader is closed. please build again");
+    }
+    return rowCountInSplits.get(rowCountInSplits.size() - 1);
+  }
+
+  /**
+   * This interface is for python to call java.
+   * Because python cannot understand java Long object. so send string object.
+   *
+   * Get total rows in the folder.
+   * It is based on the snapshot of files taken while building the reader.
+   *
+   *
+   * @return total rows from all the files in the reader.
+   */
+  public String getTotalRowsAsString() {
+    if (isClosed) {
+      throw new RuntimeException("Pagination Reader is closed. please build again");
+    }
+    return (rowCountInSplits.get(rowCountInSplits.size() - 1)).toString();
+  }
+
+  private static int findBlockletIndex(List<Long> summationArray2, Long key) {
+    // summation array is in sorted order, so can use the binary search.
+    int index = Collections.binarySearch(summationArray2, key);
+    if (index < 0) {
+      // when key not found, binary search returns negative index [-1 to -N].
+      // which is the possible place where key can be inserted.
+      // with one shifted position. As 0 is also a valid index.
+      index = index + 1; // offset the one index shifted.
+      index = Math.abs(index);
+    }
+    return index;
+  }
+
+  private int[] getBlockletIndexRange(long from, long to) {
+    // find the matching blocklet index range by checking with from and to.
+    int upperBound = findBlockletIndex(rowCountInSplits, to);
+    // lower bound cannot be more than upper bound, so work on sub list.
+    int lowerBound = findBlockletIndex(rowCountInSplits.subList(0, upperBound), from);
+    return new int[] {lowerBound, upperBound};
```

Comment:
suggestion: create new class and return a object of this class

Reply:
done